tests/unit/test.config.js

```diff
@@ -0,0 +1,187 @@
+/* @flow */
+import path from 'path';
+
+import {assert} from 'chai';
+import {describe, it} from 'mocha';
+import sinon from 'sinon';
+import {fs} from 'mz';
+
+import {Program} from '../../src/program';
+import {
+  applyConfigToArgv,
+  loadJSConfigFile,
+} from '../../src/config';
+import {withTempDir} from '../../src/util/temp-dir';
+import {UsageError} from '../../src/errors';
+
+type MakeArgvParams = {|
+  userCmd: Array<string>,
```

Comment:
this should be marked as optional since the implementation initializes it to a default value